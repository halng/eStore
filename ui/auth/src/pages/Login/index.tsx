import React from "react";
import { useForm, SubmitHandler } from "react-hook-form";
import { UserLogin } from "@model";
import { Link } from "react-router-dom";

const LogIn = () => {
  const { register, handleSubmit } = useForm<UserLogin>();

  const onSubmitHandler: SubmitHandler<UserLogin> = (data) => {
    console.log(data);
  };

  return (
    <div className="bg-gray-50 dark:bg-gray-900">
      <div className="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0">
        <a
          href="#"
          className="flex items-center mb-6 text-2xl font-semibold text-gray-900 dark:text-white"
        >
          <img
            className="w-8 h-8 mr-2"
            src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/logo.svg"
            alt="logo"
          />
          eStore
        </a>
        <div className="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700">
          <div className="p-6 space-y-4 md:space-y-6 sm:p-8">
            <h1 className="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white">
              Login
            </h1>
            <form
              className="space-y-4 md:space-y-6"
              onSubmit={handleSubmit(onSubmitHandler)}
            >
              <div>
                <label
                  htmlFor="email"
                  className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                >
                  Email/Username
                </label>
                <input
                  type="text"
                  {...register("username")}
                  id="email"
                  className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  placeholder="Enter email or username"
                />
              </div>
              <div>
                <label
                  htmlFor="password"
                  className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                >
                  Password
                </label>
                <input
                  type="password"
                  {...register("password")}
                  id="password"
                  placeholder="••••••••"
                  className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                />
              </div>
              <button
                type="submit"
                className={`w-full text-white focus:ring-4  bg-primary-600 hover:bg-primary-700 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800`}
              >
                Login
              </button>
              <p className="text-sm font-light text-gray-500 dark:text-gray-400">
                Forgot password?{" "}
                <Link
                  to="/auth/forgot-password"
                  className="font-medium text-primary-600 hover:underline dark:text-primary-500"
                >
                  Click here
                </Link>
              </p>
              <p className="text-sm font-light text-gray-500 dark:text-gray-400">
                Not have an account?{" "}
                <Link
                  to="/auth/register"
                  className="font-medium text-primary-600 hover:underline dark:text-primary-500"
                >
                  Register here
                </Link>
              </p>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default LogIn;
