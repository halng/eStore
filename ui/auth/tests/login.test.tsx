import "@testing-library/jest-dom";
import Login from "../src/pages/Login";
import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Test form rendering", () => {
  it("Render Login component", () => {
    const loginComponent = renderer.create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    let tree = loginComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
