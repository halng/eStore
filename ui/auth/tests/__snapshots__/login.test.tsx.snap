// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test form rendering Render Login component 1`] = `
<div
  className="bg-gray-50 dark:bg-gray-900"
>
  <div
    className="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0"
  >
    <a
      className="flex items-center mb-6 text-2xl font-semibold text-gray-900 dark:text-white"
      href="#"
    >
      <img
        alt="logo"
        className="w-8 h-8 mr-2"
        src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/logo.svg"
      />
      eStore
    </a>
    <div
      className="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
    >
      <div
        className="p-6 space-y-4 md:space-y-6 sm:p-8"
      >
        <h1
          className="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
        >
          Login
        </h1>
        <form
          className="space-y-4 md:space-y-6"
          onSubmit={[Function]}
        >
          <div>
            <label
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              htmlFor="email"
            >
              Email/Username
            </label>
            <input
              className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              id="email"
              name="username"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter email or username"
              type="text"
            />
          </div>
          <div>
            <label
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="••••••••"
              type="password"
            />
          </div>
          <button
            className="w-full text-white focus:ring-4  bg-primary-600 hover:bg-primary-700 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
            type="submit"
          >
            Login
          </button>
          <p
            className="text-sm font-light text-gray-500 dark:text-gray-400"
          >
            Forgot password?
             
            <a
              className="font-medium text-primary-600 hover:underline dark:text-primary-500"
              href="/auth/forgot-password"
              onClick={[Function]}
            >
              Click here
            </a>
          </p>
          <p
            className="text-sm font-light text-gray-500 dark:text-gray-400"
          >
            Not have an account?
             
            <a
              className="font-medium text-primary-600 hover:underline dark:text-primary-500"
              href="/auth/register"
              onClick={[Function]}
            >
              Register here
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test welcome rendering Render Login component 1`] = `
<div
  className="bg-gray-50 dark:bg-gray-900"
>
  <div
    className="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0"
  >
    <a
      className="flex items-center mb-6 text-2xl font-semibold text-gray-900 dark:text-white"
      href="#"
    >
      <img
        alt="logo"
        className="w-8 h-8 mr-2"
        src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/logo.svg"
      />
      eStore
    </a>
    <div
      className="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
    >
      <div
        className="p-6 space-y-4 md:space-y-6 sm:p-8"
      >
        <h1
          className="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
        >
          Login
        </h1>
        <form
          className="space-y-4 md:space-y-6"
          onSubmit={[Function]}
        >
          <div>
            <label
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              htmlFor="email"
            >
              Email/Username
            </label>
            <input
              className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              id="email"
              name="username"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter email or username"
              type="text"
            />
          </div>
          <div>
            <label
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="••••••••"
              type="password"
            />
          </div>
          <button
            className="w-full text-white focus:ring-4  bg-primary-600 hover:bg-primary-700 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
            type="submit"
          >
            Login
          </button>
          <p
            className="text-sm font-light text-gray-500 dark:text-gray-400"
          >
            Forgot password?
             
            <a
              className="font-medium text-primary-600 hover:underline dark:text-primary-500"
              href="/auth/forgot-password"
              onClick={[Function]}
            >
              Click here
            </a>
          </p>
          <p
            className="text-sm font-light text-gray-500 dark:text-gray-400"
          >
            Not have an account?
             
            <a
              className="font-medium text-primary-600 hover:underline dark:text-primary-500"
              href="/auth/register"
              onClick={[Function]}
            >
              Register here
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
